pkg/verifier/aws/aws_verifier.go

```diff
@@ -318,3 +324,120 @@ func (a *AwsVerifier) writeDebugLogs(log string) {
 	a.Output.AddDebugLogs(log)
 	a.Logger.Debug(context.TODO(), log)
 }
+
+// CreateSecurityGroup creates a security group with the specified name and cluster tag key in a specified VPC
+func (a *AwsVerifier) CreateSecurityGroup(ctx context.Context, name, vpcId string) (*ec2.CreateSecurityGroupOutput, error) {
+	input := &ec2.CreateSecurityGroupInput{
+		GroupName:   awsTools.String(name),
+		VpcId:       &vpcId,
+		Description: awsTools.String("osd-network-verifier security group"),
+	}
+	output, err := a.AwsClient.CreateSecurityGroup(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	a.Logger.Info(ctx, "Created security group with ID: %s", *output.GroupId)
+
+	input_rules := &ec2.AuthorizeSecurityGroupEgressInput{
+		GroupId: output.GroupId,
+		IpPermissions: []ec2Types.IpPermission{
+			{
+				FromPort:   awsTools.Int32(80),
+				ToPort:     awsTools.Int32(80),
+				IpProtocol: awsTools.String("tcp"),
+				IpRanges: []ec2Types.IpRange{
+					{
+						CidrIp: awsTools.String("0.0.0.0/0"),
+					},
+				},
+			},
+			{
+				FromPort:   awsTools.Int32(443),
+				ToPort:     awsTools.Int32(443),
+				IpProtocol: awsTools.String("tcp"),
+				IpRanges: []ec2Types.IpRange{
+					{
+						CidrIp: awsTools.String("0.0.0.0/0"),
+					},
+				},
+			},
+			{
+				FromPort:   awsTools.Int32(9997),
+				ToPort:     awsTools.Int32(9997),
+				IpProtocol: awsTools.String("tcp"),
+				IpRanges: []ec2Types.IpRange{
+					{
+						CidrIp: awsTools.String("0.0.0.0/0"),
+					},
+				},
+			},
+		},
+	}
+
+	if _, err := a.AwsClient.AuthorizeSecurityGroupEgress(ctx, input_rules); err != nil {
+		return nil, err
+	}
+
+	revoke_default_egress := &ec2.RevokeSecurityGroupEgressInput{
+		GroupId: output.GroupId,
+		IpPermissions: []ec2Types.IpPermission{
+			{
+				FromPort:   awsTools.Int32(-1),
+				ToPort:     awsTools.Int32(-1),
+				IpProtocol: awsTools.String("-1"),
+				IpRanges: []ec2Types.IpRange{
+					{
+						CidrIp: awsTools.String("0.0.0.0/0"),
+					},
+				},
+			},
+		},
+	}
+
+	if _, err := a.AwsClient.RevokeSecurityGroupEgress(ctx, revoke_default_egress); err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+// DeleteSecurityGroup deletes a security group with the specified ID
```

Comment:
We can put this right in the main logic. Its just setting up inputs and calling the client . Since we abstract it away in the awclient package the "DeleteSecruityGroups" function we don't need this abstraction in here for testing either.  Also is the function used ?

Reply:
It's used ~line 105 of pkg/verifier/aws/entry_point.go

Reply:
@mjlshen @anispate I think we should still remove this func unnecessary abstraction just call it directly in the entry_point with the client.